### DISCUSSION TITLE
Access API via url. Any solution?

### DISCUSSION THREAD

**Discussion by Booker141** [vercel/next.js#51773] [Help] [answered]
### Summary

Hello I would like to ask you a question about Next.js 12 that I have not found anywhere and I hope you can help me. I am deploying an application that uses developed endpoints in the /pages/api folder, how can I prevent the user from accessing the raw json via the url? Example: user accesses app.vercel.app/api/posts and all the data appears. I await your response and if possible. Thanks in advance.

**Comment by tj-noor** [comment]
To make it difficult to access from a web browser, you could require a specific header (e.g. Authorization, x-abc-key, etc), or, convert it from what I presume a GET call to a POST. But it could have implications with your caching strategy.

What's your goal behind preventing access to the json from that endpoint? If your web app is able to access the APIs from the client, any user can access it if they simply open their dev tools.

**Reply by Booker141** [reply]
Thank you for the answer! I feel it is unsecure if users have access to API from client right? I mean I don't have a lot of info about this.

**Reply by tj-noor** [reply, marked as answer]
Is the API made for the client site to populate something on its UI? If so, the client/user will need/have access to it regardless.

If you don't have anything the user shouldn't see, then having a public api is a perfectly valid use case.

Otherwise, you need to add safe guards such as authentication, encryption, and/or limit the data that the API returns.

**Reply by Booker141** [reply]
Okay thank you so much!!
